outbox-core/src/main/java/io/praesidio/outbox/Outbox.java

```diff
@@ -0,0 +1,30 @@
+package io.praesidio.outbox;
+
+import io.praesidio.outbox.values.MessageType;
+import lombok.NonNull;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class Outbox {
+
+    private final MessageRepository messageRepository;
+    private final Map<MessageType, MessageSerializer> serializers;
+
+    public Outbox(@NonNull MessageRepository messageRepository,
+                  @NonNull Collection<MessageSerializer> serializers) {
```

Comment:
Collection or Set?

Reply:
I don't think we care about the `Collection` type here - library user could pass anything, we need to iterate the whole collection anyway. I would leave the `Collection` interface here if you don't mind.

Reply:
I would also change the rest of such places to use `Collection`.

Reply:
Because I agree that there is inconsistency now.